sharktank/tests/ops/sharded_test.py

```diff
@@ -887,6 +888,38 @@ def testReplicatedLhsAndRhs(self):
             assert ops.equal(shard, unsharded_result)
 
 
+@parameterized_class(
+    ("keepdim", "mean_dim_delta"), list(itertools.product([True, False], [-1, 0, +1]))
+)
+class MeanTest(unittest.TestCase):
+    def setUp(self):
+        self.shape = (2, 4, 6)
+        self.shard_dim = 1
+        self.mean_dim = self.shard_dim + self.mean_dim_delta
+        self.shard_count = 2
+
+    def testMeanReplicated(self):
+        tensor = torch.rand(self.shape, dtype=torch.float32)
+        expected_result = ops.mean(tensor, dim=self.mean_dim, keepdim=self.keepdim)
+        actual_result = ops.mean(
+            ops.replicate(tensor, count=self.shard_count),
+            dim=self.mean_dim,
+            keepdim=self.keepdim,
+        )
+        assert ops.equal(expected_result, actual_result)
```

Comment:
If for some reason we ever change the order of additions compared to the reference this will fail due to rounding loss of precision. It is better to do `torch.testing.assert_close` to allow for some error tolerance.
The same holds for the other test.